src/git-repository.coffee

```diff
@@ -79,6 +79,9 @@ class GitRepository
     unless @repo?
       throw new Error("No Git repository found searching path: #{path}")
 
+    @watch = fs.watch path, { persistent: false, recursive: true }, (event, file) =>
```

Comment:
According to the node docs:

> The fs.watch API is not 100% consistent across platforms, and is unavailable in some situations.
>
> The recursive option is currently supported on OS X. Only FSEvents supports this type of file watching so it is unlikely any additional platforms will be added soon.

So this won't be enough on Linux and Windows and so the focus handler should probably be kept.